odoo_2_odoo_data_transfer/models/odoo_data_transfer_log.py

```diff
@@ -67,24 +61,58 @@ class OdooDataTransferTemplateLog(models.Model):
         domain=[("error_type", "=", "other_error")],
     )
     transfered_records_counter = fields.Integer(
-        compute="_compute_transfered_records_counter"
+        compute="_compute_transfered_records_counter",
+        store=True,
     )
     record_total_counter = fields.Integer()
 
+    first_transferred_id = fields.Integer(
+        compute="_compute_transfered_records_counter",
+        store=True,
+    )
+    last_transferred_id = fields.Integer(
+        compute="_compute_transfered_records_counter",
+        store=True,
+    )
+    is_failed_migration = fields.Boolean(readonly=True)
+
     @api.depends("transfered_records_ids")
     def _compute_transfered_records_counter(self):
         for rec in self:
             rec.transfered_records_counter = len(rec.transfered_records_ids)
+            rec.first_transferred_id = (
+                min(rec.transfered_records_ids.mapped("remote_id"))
+                if rec.transfered_records_ids
+                else 0
+            )
+            rec.last_transferred_id = (
+                max(rec.transfered_records_ids.mapped("remote_id"))
+                if rec.transfered_records_ids
+                else 0
+            )
 
     @api.model
-    def _get_transfered_record_log_lines(self, model_id):
+    def _get_last_transfered_record_id(self, model_id):
         """Returns record of model created befor with this module"""
-        return (
-            self.search([("local_target_model_id", "=", model_id.id)])
-            .mapped("transfered_records_ids")
-            .filtered(lambda li: li.local_id and li.local_id.exists())
+        return max(
+            self.search([("local_target_model_id", "=", model_id.id)]).mapped(
```

Comment:
If I'm not mistaken, it would be faster to perform a search with desc or asc order based on the remote_id field and a return limit of 1.

Reply:
if use this method of max/min a lot, create a function for this .

Reply:
Okay, i will change it, but i don't this is only used one time per transference and the speed increment would not be significant @ValentinVinagre

Reply:
Keep in mind that with mapping, the system has to gather all that information from the database and then load it into a list and search for the maximum amount. With search, the system does all the searching for you. The improvement should be noticeable.

Reply:
Changes applied @ValentinVinagre